Makefile

```diff
@@ -81,13 +81,13 @@ PKG_TEST_DIRS := $(filter-out $(INTEG_TEST_ROOT)%,$(TEST_DIRS))
 #   Packages are specified as import paths.
 GOCOVERPKG_ARG := -coverpkg="$(PROJECT_ROOT)/common/...,$(PROJECT_ROOT)/service/...,$(PROJECT_ROOT)/client/...,$(PROJECT_ROOT)/tools/..."
 
-dep-ensured:
-	./install-dep.sh
-	dep ensure
+go-mod:
+	go mod tidy
+	go mod vendor
```

Comment:
This is only necessary if you want to vendor dependencies and commit the vendor directory. Since you are not doing it with dep, I'm guessing you don't really need it.

Reply:
ah, that is true, previously the yarpc was using something from vendor dir
```
yarpc-install:
	go get './vendor/go.uber.org/thriftrw'
	go get './vendor/go.uber.org/yarpc/encoding/thrift/thriftrw-plugin-yarpc'
```
basically the orioginal code was trying to use the same version of thriftrw and thriftrw-plugin-yarpc for code generation.

the thing is, `go get './vendor/go.uber.org/thriftrw'` can work
however `go get './vendor/go.uber.org/yarpc/encoding/thrift/thriftrw-plugin-yarpc'` cannot, due to missing dir (probably due to something similar to https://github.com/golang/go/issues/32504).

any suggestion?

Reply:
If you want to make sure that the same version is used for both libraries, you should probably use the [replace directive](https://github.com/golang/go/wiki/Modules#when-should-i-use-the-replace-directive)

Reply:
that is not what I meant, not version control.

the thriftrw and thriftrw-plugin-yarpc are both tool dependency, and previously `go.uber.org/yarpc/encoding/thrift/thriftrw-plugin-yarpc` cannot be successfully imported due to it being a `main package`

I think the correct solution is `tools dependency`

Reply:
Ah, okay, got it.

I would consider using a tool like [gobin](https://github.com/myitcv/gobin) then. (I'm personally not a huge fan of the tool dependency pattern)

Here is an example: https://github.com/sagikazarmark/modern-go-application/blob/master/Makefile#L186-L202